server/src/main/java/org/opensearch/index/translog/transfer/TranslogTransferManager.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.translog.transfer;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.message.ParameterizedMessage;
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.LatchedActionListener;
+import org.opensearch.common.blobstore.BlobPath;
+import org.opensearch.index.translog.transfer.listener.FileTransferListener;
+import org.opensearch.index.translog.transfer.listener.TranslogTransferListener;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.UnaryOperator;
+import java.util.stream.Collectors;
+
+import static org.opensearch.index.translog.transfer.FileSnapshot.TransferFileSnapshot;
+import static org.opensearch.index.translog.transfer.FileSnapshot.TranslogFileSnapshot;
+
+/**
+ * The class responsible for orchestrating the transfer of a {@link TransferSnapshot} via a {@link TransferService}
+ *
+ * @opensearch.internal
+ */
+public class TranslogTransferManager {
+
+    private final TransferService transferService;
+    private final BlobPath remoteBaseTransferPath;
+    private final FileTransferListener fileTransferListener;
+    private final UnaryOperator<Set<TransferFileSnapshot>> exclusionFilter;
+
+    private static final long TRANSFER_TIMEOUT_IN_MILLIS = 30000;
+
+    private static final Logger logger = LogManager.getLogger(TranslogTransferManager.class);
+
+    public TranslogTransferManager(
+        TransferService transferService,
+        BlobPath remoteBaseTransferPath,
+        FileTransferListener fileTransferListener,
+        UnaryOperator<Set<TransferFileSnapshot>> exclusionFilter
+    ) {
+        this.transferService = transferService;
+        this.remoteBaseTransferPath = remoteBaseTransferPath;
+        this.fileTransferListener = fileTransferListener;
+        this.exclusionFilter = exclusionFilter;
+    }
+
+    public boolean transferSnapshot(TransferSnapshot transferSnapshot, TranslogTransferListener translogTransferListener)
```

Comment:
should we return `boolean` from here ? we are already taking a listener here and the return value would be ignored by the caller.

Reply:
This listener is not meant for a call back but a means to update transfer tracking